README.md

```diff
@@ -8,21 +8,179 @@
 
 ## Overview
 
-[//]: # (TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details.)
+[HubSpot](https://www.hubspot.com/) is an AI-powered customer relationship management (CRM) platform.
+
+The `ballerinax/module-ballerinax-hubspot.crm.engagement.notes` connector offers APIs to connect and interact with the [Hubspot Engagement Notes API](https://developers.hubspot.com/docs/reference/api/crm/engagements/notes) endpoints, specifically based on the [HubSpot REST API](https://developers.hubspot.com/docs/reference/api/overview).
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+### Step 1: Create/Login to a HubSpot Developer Account
+
+If you already have a HubSpot Developer Account, go to the [HubSpot developer portal](https://app.hubspot.com/).
+
+If you don't have an account, you can sign up to a free account [here](https://developers.hubspot.com/get-started).
+
+### Step 2 (Optional): Create a [Developer Test Account](https://developers.hubspot.com/beta-docs/getting-started/account-types#developer-test-accounts) under your account
```

Comment:
During the review it was discussed to move the link to the description. Please fix the other places as well.